### DISCUSSION TITLE
How to display coloured chapter ranges?

### DISCUSSION THREAD

**Discussion by Zabooby** [tomasklaen/uosc#408] [Q&A] [answered]
Currently my chapter ranges look like this:

![image](https://user-images.githubusercontent.com/78969986/210077102-d4f2e511-ff3a-41ab-b640-53d7c179d339.png)

How do I add colour to them, here is my [uosc.conf ](https://github.com/Zabooby/mpv-config/blob/main/portable_config/script-opts/uosc.conf)file.

**Comment by tomasklaen** [comment, marked as answer]
Currently only openings, endings, intros, outros, and ads chapter indicators are converted into ranges. This is done by internal range detection which can be "localized" with `chapter_range_patterns` option.

If your chapters are not one of the supported ranges, and it makes sense, we might add a new range type.

If the chapters _are_ one of the supported range types, but they are named like `chapter 1`, `chapter 2`, then they'll never be converted into a range as their names are not enough for us to know if, and what kind of range they are.

**Reply by Zabooby** [reply]
So it basically depends on the file using supported chapter ranges. 

Thanks for the answer!

**Reply by xfzv** [reply]
> If your chapters are not one of the supported ranges, and it makes sense, we might add a new range type.

~~It would be cool if we could add custom ranges.~~

Edit: never mind, it's already possible with `chapter_ranges` and `chapter_range_patterns`:

https://github.com/tomasklaen/uosc/blob/ed58830ca781f1db3564f25f4226b7eeb11cf036/script-opts/uosc.conf#L202-L206
